### PR TITLE
Adds a toy llama2 model

### PR DESCRIPTION
**Pull request by wang2yn84**
# Description

The smallest llama model we support is 7b, which is still too slow for local development. This PR creates a toy model with much smaller For functional testing and non accuracy validation only.

# Tests
export XLA_FLAGS="--xla_gpu_enable_latency_hiding_scheduler=true --xla_gpu_enable_command_buffer=FUSION --xla_disable_hlo_passes=rematerialization --xla_disable_hlo_passes=gpu-convert-async-collectives-to-sync" # flags from NVidia
export TF_FORCE_GPU_ALLOW_GROWTH=true
export BASE_OUTPUT_DIRECTORY=/scratch/temp
export ASYNC_CHECKPOINTING=false
export XLA_PYTHON_CLIENT_MEM_FRACTION=0.92
export PER_DEVICE_BATCH_SIZE=2

python3  MaxText/decode.py MaxText/configs/base.yml base_output_directory=${BASE_OUTPUT_DIRECTORY}  model_name='llama2-toy' max_prefill_predict_length=1024  max_target_length=2048  attention=dot_product scan_layers=false hardware=gpu async_checkpointing=${ASYNC_CHECKPOINTING} per_device_batch_size=${PER_DEVICE_BATCH_SIZE} run_name=$(date +%Y-%m-%d-%H-%M) ici_fsdp_parallelism=1 ici_autoregressive_parallelism=1 ici_tensor_parallelism=-1 skip_jax_distributed_system=True weight_dtype=float16 dtype=float16 kv_quant_dtype=fp8 quantize_kvcache=True

Finish less than 1 minute.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [ ] I have performed a self-review of my code.
- [ ] I have necessary comments in my code, particularly in hard-to-understand areas.
- [ ] I have run end-to-end tests tests and provided workload links above if applicable.
- [ ] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

**Review by richjames0 — APPROVED** [review]
lgtm

**Comment by gobbleturk** [comment]
Curious how this script is useful to you? You can already run this model by setting the dimensions manually (e.g. base_emb_dim=xx). Is this just to avoid setting all of these parameters all of the time?

File: MaxText/configs/models/llama2-toy.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# model config for llama2-44m
+
+base_emb_dim: 256
+base_num_query_heads: 64
+base_num_kv_heads: 8
+base_mlp_dim: 896
```
Comment (gobbleturk) [inline MaxText/configs/models/llama2-toy.yml:20]: nit: why not 1024?

File: MaxText/maxengine.py [hunk 2]
```diff
@@ -398,7 +398,7 @@ def prefill_aot(  # pylint: disable=too-many-positional-arguments
 
   @functools.partial(jax.jit, static_argnums=(0,), static_argnames=("request_id",))
   def _prefill_jit(
-      self, # pytype: disable=signature-mismatch
+      self,  # pytype: disable=signature-mismatch
```
Comment (gobbleturk) [inline MaxText/maxengine.py:401]: Please try to avoid random whitespace changes in unrelated files - unless this is necessary for our linter?

Comment (bvandermoon) [reply MaxText/maxengine.py:401]: This is necessary for the linter based on this [PR 1514 comment](https://github.com/AI-Hypercomputer/maxtext/pull/1514/files#r2027736652). Not exactly sure why though
